tests/unit/test_skiplist_generic.c

```diff
@@ -106,20 +106,21 @@ void test_numeric_sorts(void) {
   // vals are defined as the order the keys should be in after sorting
   double vals[] = {5, 2, 0, 6, 3, 4, 1};
 
-  skiplist *numeric_sl = skiplistCreate(cmpDoubleKeys, NULL, cmpDoubleVals, NULL);
+  skiplist *numeric_sl = skiplistCreate(cmpDoubleKeys, NULL, cmpDoubleVals, NULL, NULL);
 
   for (int i = 0; i < 7; i ++) {
     skiplistInsert(numeric_sl, keys + i, vals + i);
   }
 
   double lastval = -1;
   double *retval;
-  skiplistIterator iter = skiplistIterateAll(numeric_sl);
+  skiplistIterator *iter = skiplistIterateAll(numeric_sl);
   // Verify that skiplist is sorted on numeric keys
-  while ((retval = skiplistIterator_Next(&iter)) != NULL) {
+  while ((retval = skiplistIterator_Next(iter)) != NULL) {
     assert(lastval + 1 == *(double *)retval);
     lastval = *retval;
   }
+  free(iter);
```

Comment:
Usually we have a dedicated function for freeing "objects"
I think it would be better to have both reset and free functions:
1. skiplistIterate_Reset
2. skiplistIterate_Free